### PR TITLE
CIV-11938 bug task list

### PR DESCRIPTION
**Pull request by GorilaNaranja**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-11967
https://tools.hmcts.net/jira/browse/CIV-11938
https://tools.hmcts.net/jira/browse/CIV-11964

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by neeta-hmcts — APPROVED** [review]

File: src/main/services/features/claimantResponse/claimantResponseService.ts [hunk 1]
```diff
@@ -69,13 +69,65 @@ const saveClaimantResponse = async (claimId: string, value: any, claimantRespons
...
+function resetTaskListData(claim: Claim, claimantResponsePropertyName: string, parentPropertyName?: string) {
+  if (isAcceptOrRejectTheAmountSubmitted(claimantResponsePropertyName)) {
+    delete claim.claimantResponse.hasPartPaymentBeenAccepted;
+    delete claim.claimantResponse.mediation;
+    delete claim.claimantResponse.directionQuestionnaire;
+    delete claim.claimantResponse.fullAdmitSetDateAcceptPayment;
+    deleteTaskListData(claim);
+  } else if (isAcceptOrRejectRepaymentPlanSubmitted(claimantResponsePropertyName)) {
+    deleteTaskListData(claim);
+  } else if (isProposeAnAlternativeRepaymentPlanSubmitted(claimantResponsePropertyName,parentPropertyName )) {
```
Comment (paul-pearson) [inline src/main/services/features/claimantResponse/claimantResponseService.ts:89]: Could fix formatting - whitespace after comma here.

File: src/main/services/features/claimantResponse/claimantResponseService.ts [hunk 2]
```diff
@@ -69,13 +69,65 @@ const saveClaimantResponse = async (claimId: string, value: any, claimantRespons
...
+    delete claim.claimantResponse.chooseHowToProceed;
+    deleteTaskListFormaliseRepaymentData(claim);
+  } else if (isChooseHowToProceedSubmitted(claimantResponsePropertyName , parentPropertyName)) {
```
Comment (paul-pearson) [inline src/main/services/features/claimantResponse/claimantResponseService.ts:92]: Same here - extra space before comma could be removed

**Review by paul-pearson — APPROVED** [review]

File: src/main/services/features/claimantResponse/claimantResponseService.ts (hunk 1 above)
Comment (jeswanth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseService.ts]: fixed specified formatting issues

File: src/main/services/features/claimantResponse/claimantResponseService.ts (hunk 2 above)
Comment (jeswanth-hmcts) [reply src/main/services/features/claimantResponse/claimantResponseService.ts]: fixed specified formatting issues

**Review by kalachandrasekar1 — APPROVED** [review]

**Pull request merged**